src/NuGet.Server.Core/Infrastructure/KnownPathUtility.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+
+namespace NuGet.Server.Core.Infrastructure
+{
+    public static class KnownPathUtility
+    {
+        /// <summary>
+        /// Determines if the file path could have been generated by <see cref="ExpandedPackageRepository"/>.
+        /// </summary>
+        /// <param name="path">The file path to parse.</param>
+        /// <param name="id">The package ID found.</param>
+        /// <param name="version">The package version found.</param>
+        /// <returns>True if the file name is known.</returns>
+        public static bool TryParseFileName(string path, out string id, out SemanticVersion version)
+        {
+            id = null;
+            version = null;
+
+            if (path == null)
+            {
+                return false;
+            }
+
+            var pathPieces = path.Split(Path.DirectorySeparatorChar);
```

Comment:
so, this will not work if `/` is used for directory separator?..
We really should get into habit of using `DirectoryInfo` and `FileInfo` for directory and file references, it would handle both types of separators transparently...

Reply:
File system events use the OS's directory separator so we can depend on this.